app/src/main/java/com/example/marvelapp/data/mapper/CharacterRealmMapper.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.marvelapp.data.mapper
+
+import com.example.marvelapp.data.local.model.CharacterRealmEntity
+import com.example.marvelapp.data.local.model.ComicsRealmEntity
+import com.example.marvelapp.data.local.model.EventsRealmEntity
+import com.example.marvelapp.data.local.model.SeriesRealmEntity
+import com.example.marvelapp.data.local.model.StoriesRealmEntity
+import com.example.marvelapp.data.local.model.ThumbnailRealmEntity
+import com.example.marvelapp.data.local.model.UrlRealmEntity
+import com.example.marvelapp.domain.entity.ComicsEntity
+import com.example.marvelapp.domain.entity.EventsEntity
+import com.example.marvelapp.domain.entity.SeriesEntity
+import com.example.marvelapp.domain.entity.StoriesEntity
+import com.example.marvelapp.domain.entity.ThumbnailEntity
+import com.example.marvelapp.domain.entity.UrlsEntity
+import io.realm.RealmList
+
+object CharacterRealmMapper {
+
+    private fun transformDBCharacter(characterEntity: com.example.marvelapp.domain.entity.CharacterEntity): CharacterRealmEntity =
+        CharacterRealmEntity(
+            characterEntity.id,
+            characterEntity.name,
+            characterEntity.description,
+            characterEntity.modified,
+            transformDBThumbnail(characterEntity.thumbnail),
+            characterEntity.resourceUri,
+            transformDBComics(characterEntity.comics),
+            transformDBEvents(characterEntity.events),
+            transformDBStories(characterEntity.stories),
+            transformDBSeries(characterEntity.series),
+            transformDBListOfUrls(characterEntity.urls) as RealmList<UrlRealmEntity>
+        )
+
+    fun transformDBListOfUrls(urlsEntity: List<UrlsEntity>): List<UrlRealmEntity> {
```

Comment:
cant you return a RealmList here and avoid de casting on line 32?

Reply:
Thx, for the advice, now its done